src/modules/dashboard/sagas/actions/removeExistingSafes.ts

```diff
@@ -0,0 +1,221 @@
+  getTxChannel,
+} from '../../../core/sagas';
+import { ipfsUpload } from '../../../core/sagas/ipfs';
+import {
+  transactionReady,
+  transactionPending,
+  transactionAddParams,
+} from '../../../core/actionCreators';
+
+function* removeExistingSafesAction({
+  payload: { colonyName, colonyAddress, safeAddresses, annotationMessage },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_ACTION_REMOVE_EXISTING_SAFES>) {
+  let txChannel;
+  try {
+    const apolloClient = TEMP_getContext(ContextModule.ApolloClient);
+    const ipfsWithFallback = TEMP_getContext(ContextModule.IPFSWithFallback);
+
+    if (isEmpty(safeAddresses)) {
+      throw new Error('A list with safe addresses is required to remove safes');
+    }
+
+    txChannel = yield call(getTxChannel, metaId);
+    const batchKey = 'removeExistingSafes';
+    const {
+      removeExistingSafesAction: removeExistingSafes,
+      annotateRemoveExistingSafesAction: annotateRemoveExistingSafes,
+    } = yield createTransactionChannels(metaId, [
+      'removeExistingSafesAction',
+      'annotateRemoveExistingSafesAction',
+    ]);
+
+    const createGroupTransaction = ({ id, index }, config) =>
+      fork(createTransaction, id, {
+        ...config,
+        group: {
+          key: batchKey,
+          id: metaId,
+          index,
+        },
+      });
+
+    yield createGroupTransaction(removeExistingSafes, {
+      context: ClientType.ColonyClient,
+      methodName: 'editColony',
+      identifier: colonyAddress,
+      params: [],
+      ready: false,
+    });
+
+    if (annotationMessage) {
+      yield createGroupTransaction(annotateRemoveExistingSafes, {
+        context: ClientType.ColonyClient,
+        methodName: 'annotateTransaction',
+        identifier: colonyAddress,
+        params: [],
+        ready: false,
+      });
+    }
+
+    yield takeFrom(
+      removeExistingSafes.channel,
+      ActionTypes.TRANSACTION_CREATED,
+    );
+    if (annotationMessage) {
+      yield takeFrom(
+        annotateRemoveExistingSafes.channel,
+        ActionTypes.TRANSACTION_CREATED,
+      );
+    }
+
+    yield put(transactionPending(removeExistingSafes.id));
+
+    /*
+     * Fetch colony data from the subgraph
+     * And destructure the metadata hash.
+     */
+
+    const {
+      data: {
+        colony: { metadata: currentMetadataIPFSHash },
+      },
+    } = yield apolloClient.query<
+      SubgraphColonyQuery,
+      SubgraphColonyQueryVariables
+    >({
+      query: SubgraphColonyDocument,
+      variables: {
+        address: colonyAddress.toLowerCase(),
+      },
+      fetchPolicy: 'network-only',
+    });
+
+    const currentMetadata = yield call(
+      ipfsWithFallback.getString,
+      currentMetadataIPFSHash,
+    );
+
+    const colonyMetadata = JSON.parse(currentMetadata);
+
+    const updatedColonySafes = colonyMetadata.colonySafes.filter((safe) =>
+      safeAddresses.find((safeAddress) => safeAddress !== safe.contractAddress),
+    );
```

Comment:
For more than one safe in the `safeAddresses` array, the find predicate will always return true and therefore the filter will return the same array. If there is more than one safe in `safeAddresses`, you will always be able to find a safe address that is not equal to the current item in the filter. 

I suggest something like this, which is saying, if there is a safe in the metadata that is also in `safeAddresses`, don't return it in the updated array. 

``` js
const updatedColonySafes = colonyMetadata.colonySafes.filter(
      (safe) =>
        !safeAddresses.some(
          (safeAddress) => safeAddress === safe.contractAddress,
        ),
    );
```

Reply:
You are completely right! This is the solution. When I tested it I remember that it was working, but it seems like I'm going crazy because, following the logic I had before, it shouldn't hahaha 

Thank you!